### PR TITLE
[IP-1033]: Bugfix: creation dates of invoices should never be changed

### PR DESCRIPTION
**Pull request by mheiduk**
## Description
only update invoice_date_due when sending email

invoice_date_created should not be updated and always stay the same

See (https://github.com/InvoicePlane/InvoicePlane/issues/1033) for more details

## Pull Request Checklist

  * [x] My code follows the code formatting guidelines.
  * [x] I have an issue ID for this pull request.
  * [x] I selected the corresponding branch.
  * [x] I have rebased my changes on top of the corresponding branch.

## Issue Type (Please check one or more)

  * [x] Bugfix
  * [x] Improvement of an existing Feature
  * [ ] New Feature

### CODE REVIEW

**Comment by nielsdrost7** [comment]
Great PR.
How do I test proper working of this PR?
What do I fill where in order to see it in action?
What should happen?
What should not happen?

**Comment by Verony-makesIT** [comment]
@mheiduk, 
Item #1036 can never be called a "bug fix" but rather, in my opinion, is a different interpretation or perspective of how **the invoice creation date _versus_ the invoice date** should be treated.
The result of your modification is that on your invoice, the invoice (creation) date may be (weeks or days) before the shipment date (official invoice date) and the due date may be more than 30 days later after that invoice date. 
This is very confusing for the customer. 
_For example:_
Period to due date = 30d
creation date: 27 dec 2023
work performed in January 2024
invoice updates: 10/01/2024 + 17/01/2024 + 28/01/2024
shipment date: 01/02/2024
_Result:_
Invoice date: **27/12/2023**
Due date: **01/03/2024**
Fiscal year: 4th quarter 2023 or 1st quarter 2024 (?)

I think tax authorities, your accountant and your client will be asking questions anyway....
Please also read my comments in item #1033.
**In any case, I cannot accept this.**

**Comment by nielsdrost7** [comment]
@mheiduk again, thank you for this PR. I'll try to go over it this weekend.
If you could just answer those questions I asked, that would be awesome.

**Comment by mheiduk** [comment]
@nielsdrost7 
> How do I test proper working of this PR? What do I fill where in order to see it in action? What should happen? What should not happen?
You have to send an overdue invoice to a client. If the setting in Settings -> Invoices -> Other settings is set to "No", then only the due date should update to the current date. If the setting is set to "Yes", then no date should be updated. The creation date should not be updated at all.

**Comment by nielsdrost7** [comment]
Thanks mam! I'll take a look this weekend

File: application/language/english/ip_lang.php [hunk 1]
```diff
@@ -332,7 +332,7 @@
     'no_overdue_invoices'                          => 'No overdue Invoices',
     'no_quotes_requiring_approval'                 => 'There are no quotes requiring approval.',
     'no_updates_available'                         => 'No updates available.',
-		'no_update_invoice_due_date_mail'							 => 'Disable the change of invoice date and due date before emailing',
+		'no_update_invoice_due_date_mail'							 => 'Disable the change of invoice due date before emailing',
```
Comment (nielsdrost7) [inline application/language/english/ip_lang.php:335]: Just the indenting, nothing special.
Also please align the arrow with the other arrows

**Review by nielsdrost7 — COMMENTED** [review]
Also:
`public function mark_sent($invoice_id)` in `Mdl_invoices.php`
Shouldn't you check for that due date as well there?

Because right now it's only checked when the invoice is emailed, but when someone manually sets the invoice to `sent` that due_date isn't checked?

What do you think?

**Comment by mheiduk** [comment]
> Also: `public function mark_sent($invoice_id)` in `Mdl_invoices.php` Shouldn't you check for that due date as well there?
> 
> Because right now it's only checked when the invoice is emailed, but when someone manually sets the invoice to `sent` that due_date isn't checked?
> 
> What do you think?

Thanks for the input! Tried to simplify it as much as possible and cleaned up some things in the process. Please check if everything is working as expected for you.

File: application/language/english/ip_lang.php (hunk 1 above)
Comment (mheiduk) [reply application/language/english/ip_lang.php]: Should be fixed

**Comment by nielsdrost7** [comment]
Great, great Job!
I merged it into our `1033` branch, as I now noticed and then merged the `1033` branch into `develop`.
I'll make sure to credit you, once we make the new release.

It was fun doing this PR with you.

**Pull request merged**